engine/internal/runtime/manager.go

```diff
@@ -61,30 +61,32 @@ func newReadyRuntime(name string, address string) runtime {
 }
 
 type runtime struct {
-	name  string
-	ready bool
+	name      string
+	ready     bool
+	errReason string
 	// address is empty when the runtime is not ready.
 	address string
 	// waitCh is used when the runtime is not ready.
 	waitCh chan struct{}
 }
 
-func (m *Manager) addRuntime(modelID string, sts appsv1.StatefulSet) (bool, error) {
+func (m *Manager) addRuntime(modelID string, sts appsv1.StatefulSet) (bool, bool, error) {
```

Comment:
How about adding a comment to explain what the first `bool` and the second `bool` mean?